### PR TITLE
[REVIEW #151 FIRST] Test all untested modules

### PR DESCRIPTION
**Pull request by jverbraeken**
- Relevant Issues: -
- Related Pull Requests: #151

* * *

## What
This Pull Request adds to the repository almost almost complete code coverage (97%) for all modules that were untested before and excludes untestable modules with an explanation

## Why
This Pull Request is needed because it's very important that all modules are tested thoroughly so that we can be sure that everything's working

## How
This feature can be viewed/tested within the project by running the tests

## Alternative implementation
Other implementations that I've have considered are -

### CODE REVIEW

File: app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java [hunk 1]
```diff
@@ -28,79 +28,93 @@ public void testCanBeInstantiatedWithMap() {
...
-        }
+    @Test(expected = IllegalArgumentException.class)
+    public void testNameNull() {
+        Map<String, String> map = new ConcurrentHashMap<>();
+        map.put("name", "foo");
+        new SimpleKey(map);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testKeyNull() {
```
Comment (TheBonheurs) [inline app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java]: Why not combine this with the one above?

File: app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java [hunk 2]
```diff
@@ -28,79 +28,93 @@ public void testCanBeInstantiatedWithMap() {
...
+        Map<String, String> map = new ConcurrentHashMap<>();
+        map.put("key", "foo");
+        new SimpleKey(map);
+    }
+
+    @Test
+    public void testGetNameReturnsNotNull() {
+        IKey key = new SimpleKey("FTP", "foobar");
+        assertNotNull(key.getName());
```
Comment (TheBonheurs) [inline app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java]: Useless test no?

File: app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java [hunk 3]
```diff
@@ -28,79 +28,93 @@ public void testCanBeInstantiatedWithMap() {
...
+    @Test
+    public void testGetNameReturnsProvidedName() {
+        IKey key = new SimpleKey("Webserver", "foobar");
+        assertEquals("Webserver", key.getName());
+    }
+
+    @Test
+    public void testGetKeyReturnsNotNull() {
+        IKey key = new SimpleKey("FTP", "foobar");
+        assertNotNull(key.getKey());
```
Comment (TheBonheurs) [inline app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java]: Same

File: app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java (hunk 2 above)
Comment (jverbraeken) [inline app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java]: Didn't write this, but it's indeed not a very useful test.

File: app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java (hunk 3 above)
Comment (jverbraeken) [inline app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java]: Jep, I don't think so either

**Pull request merged**